avocado/core/runners/avocado_instrumented.py

```diff
@@ -26,6 +26,8 @@ class and method names should be separated by a ":".  One
     """
     DEFAULT_TIMEOUT = 86400
 
+    CONFIGURATION_USED = ['run.test_parameters']
```

Comment:
I am missing a lot of values here, for example when I list all the config values used by `Test`:

```
$ grep "self._config.get(" avocado/core/test.py
        self.__sysinfo_enabled = self._config.get('sysinfo.collect.per_test',
            self.__cache_dirs = self._config.get('datadir.paths.cache_dirs')
        output_check_record = self._config.get('run.output_check_record')
        output_check = self._config.get('run.output_check')
            if not self._config.get('run.keep_tmp') and os.path.exists(
        failure_fields = self._config.get('simpletests.status.failure_fields')
            input_encoding = self._config.get('core.input_encoding')
        warn_regex = self._config.get('simpletests.status.warn_regex')
        warn_location = self._config.get('simpletests.status.warn_location')
        skip_regex = self._config.get('simpletests.status.skip_regex')
        skip_location = self._config.get('simpletests.status.skip_location')
            input_encoding = self._config.get('core.input_encoding')
```

Reply:
> I am missing a lot of values here, for example when I list all the config values used by `Test`:
> 
> ```
> $ grep "self._config.get(" avocado/core/test.py
>         self.__sysinfo_enabled = self._config.get('sysinfo.collect.per_test',
>             self.__cache_dirs = self._config.get('datadir.paths.cache_dirs')
>         output_check_record = self._config.get('run.output_check_record')
>         output_check = self._config.get('run.output_check')
>             if not self._config.get('run.keep_tmp') and os.path.exists(
>         failure_fields = self._config.get('simpletests.status.failure_fields')
>             input_encoding = self._config.get('core.input_encoding')
>         warn_regex = self._config.get('simpletests.status.warn_regex')
>         warn_location = self._config.get('simpletests.status.warn_location')
>         skip_regex = self._config.get('simpletests.status.skip_regex')
>         skip_location = self._config.get('simpletests.status.skip_location')
>             input_encoding = self._config.get('core.input_encoding')
> ```

You have actually flagged a lot of what I would call "non conformance" occurrences (thanks!).  While this pattern (having Avocado's configuration within the test class) was not something avoidable in the legacy runner, this is what we don't want to do for the nrunner.

I'll prepare a better explanation proposal to follow this PR.

Reply:
A quick follow up, looking at "simpletests" only, those:

>        failure_fields = self._config.get('simpletests.status.failure_fields')
>         warn_regex = self._config.get('simpletests.status.warn_regex')
>         warn_location = self._config.get('simpletests.status.warn_location')
>         skip_regex = self._config.get('simpletests.status.skip_regex')
>         skip_location = self._config.get('simpletests.status.skip_location')

Are never accessed when using the nrunner (exec-test).   If we do something like:

```diff
git diff
diff --git a/avocado/core/test.py b/avocado/core/test.py
index 82c7d595d..df0331e0f 100644
--- a/avocado/core/test.py
+++ b/avocado/core/test.py
@@ -1121,6 +1121,8 @@ class SimpleTest(Test):
         if self.filename is not None:
             self._command = pipes.quote(self.filename)
 
+        self._config = None
+
     @property
     def filename(self):
         """
```

And run an exec-test (nrunner):

```
$ avocado run /bin/true
JOB ID     : 66efe77a88dc8cd3612fcd54cde7b27c5643a2c1
JOB LOG    : /home/cleber/avocado/job-results/job-2021-11-01T14.01-66efe77/job.log
 (1/1) /bin/true: STARTED
 (1/1) /bin/true: PASS (0.01 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/cleber/avocado/job-results/job-2021-11-01T14.01-66efe77/results.html
JOB TIME   : 1.25 s
```

It works fine.  With the legacy runner, though:

```
$ avocado run --test-runner=runner /bin/true
JOB ID     : f5d37d4fbb636f79043b089713bf9e89bcab7dd6
JOB LOG    : /home/cleber/avocado/job-results/job-2021-11-01T13.59-f5d37d4/job.log
 (1/1) /bin/true: ERROR: 'NoneType' object has no attribute 'get' (0.05 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/cleber/avocado/job-results/job-2021-11-01T13.59-f5d37d4/results.html
JOB TIME   : 0.47 s
```

This shows the difference in approach (and features) between them.  The nrunner approach is that a "Test" class is replaced by a runner.  The runner knows about how to drive a test, but the test is the test writer's concern.

INSTRUMENTED tests are a special case due to the fact that we offer (on purpose) a Test class... but the handling of the features and configuration options needs to move into the runner IMO so the test class is left to the test writer.  In the past, we had many problems with users overwriting variables and methods, because too much "runner features" are implemented inside the class.